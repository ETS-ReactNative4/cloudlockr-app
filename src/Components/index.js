export { default as Brand } from './Brand'
export { default as InputField } from './InputField'
export { default as HorizontalLine } from './HorizontalLine'
export { default as Button } from './Button'
export { default as BasicHeader } from './BasicHeader'
export { default as DashboardHeader } from './DashboardHeader'
export { default as FileList } from './FileList'
export { default as DownloadDetail } from './DownloadDetail'
export { default as UploadDetail } from './UploadDetail'
export { default as WifiList } from './WifiList'
