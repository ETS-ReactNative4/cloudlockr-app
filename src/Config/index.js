export const Config = {
  mocking: {
    apiConnection: false,
    deviceConnection: true,
  },

  api: {
    url: 'https://cloudlockr.herokuapp.com/'
  },

  device: {
    expectedBluetoothName: 'DE1_DEVICE',                      // Expected discoverable device name
    connectionTimeout: 120,        // TODO: UPDATE            // Max # of seconds to wait for response data
    refreshRate: 3,                                           // # of seconds to wait between response polling
    maxBlobSizeBytes: 5120,        // TODO: UPDATE            // Max expected size of file data in message (must be lower than device message buffer as it does not account for other message data)
    maxTransmissionAttempts: 3                                // Max # of times the phone will reattempt transmission to device if error
  },
}
