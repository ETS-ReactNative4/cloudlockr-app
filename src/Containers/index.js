export { default as IndexStartupContainer } from './Startup/Index'
export { default as LoginContainer } from './Login/Index'
export { default as DashboardContainer } from './Dashboard/Index'
export { default as SettingsContainer } from './Settings/Index'
export { default as SettingsDeviceConnectionContainer } from './SettingsDeviceConnection/Index'
export { default as SettingsDevicePasswordContainer } from './SettingsDevicePassword/Index'
export { default as RegistrationContainer } from './Registration/Index'
export { default as SettingsDeviceWifiContainer } from './SettingsDeviceWifi/Index'
export { default as SettingsConfirmationContainer } from './SettingsConfirmation/Index'
export { default as UploadDownloadProgressContainer } from './UploadDownloadProgress/Index'
