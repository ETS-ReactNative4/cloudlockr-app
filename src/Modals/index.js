export { default as DownloadDetail } from './DownloadDetail'
export { default as UploadDetail } from './UploadDetail'
