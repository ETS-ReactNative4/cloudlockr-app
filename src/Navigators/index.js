export { default as ApplicationNavigator } from './Application'
export { default as LoginNavigator } from './Login'
export { default as DashboardNavigator } from './Dashboard'
export { default as SettingsNavigator } from './Settings'
export { default as SettingsDeviceConnectionNavigator } from './SettingsDeviceConnection'
export { default as SettingsDevicePasswordNavigator } from './SettingsDevicePassword'
export { default as SettingsDeviceWifiNavigator } from './SettingsDeviceWifi'
export { default as RegistrationNavigator } from './Registration'
export { default as SettingsConfirmationNavigator } from './SettingsConfirmation'
