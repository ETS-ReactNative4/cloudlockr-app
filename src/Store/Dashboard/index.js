import { buildSlice } from '@thecodingmachine/redux-toolkit-wrapper'
import SetDownloadInfo from '@/Store/Dashboard/SetDownloadInfo'

export default buildSlice('dashboard', [SetDownloadInfo], {
  downloadInfo: {},
}).reducer
