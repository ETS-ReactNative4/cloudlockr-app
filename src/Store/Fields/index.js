import { buildSlice } from '@thecodingmachine/redux-toolkit-wrapper'
import SetField from '@/Store/Fields/SetField'
import RemoveField from '@/Store/Fields/RemoveField'

export default buildSlice('fields', [SetField, RemoveField], {
  fields: {},
}).reducer
