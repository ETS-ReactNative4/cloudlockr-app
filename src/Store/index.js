import AsyncStorage from "@react-native-async-storage/async-storage";
import { combineReducers } from "redux";
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from "redux-persist";
import { configureStore } from "@reduxjs/toolkit";
import initialState from "@/Store/initialState";

import startup from "./Startup";
import intention from "./Intention";
import user from "./User";
import fileTransfer from "./FileTransfer";

const reducers = combineReducers({
  startup,
  intention,
  user,
  fileTransfer,
});

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
  whitelist: [],
};

const rootReducer = (state, action) => {
  // Handle special PURGE_STORE case, reset to default (does not remove local encryptionComponents)
  if (action.type === "PURGE_STORE") {
    state = initialState;
  }

  return reducers(state, action);
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => {
    const middlewares = getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    });

    if (__DEV__ && !process.env.JEST_WORKER_ID) {
      const createDebugger = require("redux-flipper").default;
      middlewares.push(createDebugger());
    }

    return middlewares;
  },
});

const persistor = persistStore(store);

const PurgeStore = () => {
  return {
    type: "PURGE_STORE",
  };
};

export { store, persistor, PurgeStore };
