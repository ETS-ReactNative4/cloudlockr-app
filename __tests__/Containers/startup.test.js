import 'react-native'
import React from 'react'
import Startup from '../../src/Containers/Startup'
import { render } from '@testing-library/react-native'

import { store } from '../../src/Store'
import { Provider } from 'react-redux'

it('renders and matches snapshot', () => {
  const { toJSON } = render(<Provider store={store}><Startup /></Provider>);
  expect(toJSON()).toMatchSnapshot();
})
