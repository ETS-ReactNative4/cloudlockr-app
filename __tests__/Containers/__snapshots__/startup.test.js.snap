// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": 90,
        "width": 260,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
